### ISSUE TITLE
OBO-level deprecation policy

### ISSUE THREAD

**Issue by jamesaoverton** [OBOFoundry/OBOFoundry.github.io#597]
_From @GoogleCodeExporter on August 12, 2015 2:4_

```
There is a GO deprecation policy and an OBI deprecation policy but there is no 
OBO-level policy. 

GO deprecation policy
http://wiki.geneontology.org/index.php/Curator_Guide:_Obsoletion

OBI deprecation policy
http://obi-ontology.org/page/OBIDeprecationPolicy

We should have an OBO level policy which can be set based on GO and OBI 
deprecation policy.
```

Original issue reported on code.google.com by `zhengj2...@gmail.com` on 11 Dec 2014 at 7:24

_Copied from original issue: OBOFoundry/Operations-Committee-RETIRED#163_

**Comment by nlharris** [comment]
what is the status of this?

**Comment by alanruttenberg** [comment]
The main issue I see is that GO allows for a URI to change even when the
old term is still valid, and allows the changing of the meaning of a term.

"ontology term creation guidelines change (for example, the development of
GO-CAM models may result in a GO term being obsoleted in favor of producing
annotations using GO-CAM that represent the same concept)"
"it is redefined in a way that invalidates existing annotations"

I don't think these should be in the OBO policy (though GO will, of course,
do what they think is best).

The OBI policy (which I can't find atm - the link given is broken), had
some extra guidance. IIRC:
Remove axioms, Place under obsolete_class, mark with owl:deprecated, add a
reason for making the term obsolete - there are some annotation properties
defined in IAO IIRC.

Here's a proposal:

MUST obsolete a term if the biology that the term represents is found to be
flawed (for example artifactual cellular components) *[taken from GO's
policy]*
MUST NOT change URIs if the denotation/meaning doesn't change.
MUST remove logical axioms
MUST NOT delete the term .
MUST place under Obsolete Class (
http://www.geneontology.org/formats/oboInOwl#ObsoleteClass)
SHOULD include annotations indicating the reason the term was made obsolete
MAY include annotations that help the user pick a new term.
Ontology developers SHOULD respect this policy

Terms used as in Key words for use in RFCs to Indicate Requirement Levels
<https://tools.ietf.org/html/rfc2119>

Alan

**Comment by matentzn** [comment]
I agree mostly with Alan, but I don't like the `MUST place under Obsolete Class`. It also contradicts the `MUST remove logical axioms`. If anything this has to be a `MAY place under Obsolete Class`. Was this discussed ever? What is the purpose of this policy (other than getting 1000ds of annoying dangling classes out of the way in the Protege class hierarchy I mean)?

**Comment by balhoff** [comment]
> The main issue I see is that GO allows for a URI to change even when the
old term is still valid, and allows the changing of the meaning of a term.

I don't really see how the GO guidelines allow either of these things. Are you (@alanruttenberg)  saying that an ontology project should be prohibited from obsoleting terms they don't want to maintain anymore? (e.g. which employ overly specific modeling that is being discouraged)

As far as "redefining" I think that is given as an example of the bad implications of changing the meaning; it invalidates the meaning of existing annotations.

**Comment by alanruttenberg** [comment]
On Tue, Jun 16, 2020 at 4:59 PM Nico Matentzoglu <notifications@github.com>
wrote:

> I agree mostly with Alan, but I don't like the MUST place under Obsolete
> Class. It also contradicts the MUST remove logical axioms. If anything
> this has to be a MAY place under Obsolete Class. Was this discussed ever?
> What is the purpose of this policy (other than getting 1000ds of annoying
> dangling classes out of the way in the Protege class hierarchy I mean)?
>
The purpose is exactly "getting 1000ds of annoying dangling classes out of
the way in the Protege class hierarchy".
I can understand that someone might object. I do think it's harmless in
this case, and removes an annoyance.
Options are to make it a SHOULD, or abandon that one, or find some other
solution that avoids the annoyances.

**Comment by alanruttenberg** [comment]
On Tue, Jun 16, 2020 at 5:08 PM Jim Balhoff <notifications@github.com>
wrote:

> The main issue I see is that GO allows for a URI to change even when the
> old term is still valid, and allows the changing of the meaning of a term.
>
> I don't really see how the GO guidelines allow either of these things. Are
> you (@alanruttenberg <https://github.com/alanruttenberg>) saying that an
> ontology project should be prohibited from obsoleting terms they don't want
> to maintain anymore? (e.g. which employ overly specific modeling that is
> being discouraged)
>
I quoted directly from the GO policy.

"ontology term creation guidelines change (for example, the development of
GO-CAM models may result in a GO term being obsoleted in favor of producing
annotations using GO-CAM that represent the same concept)"
"it is redefined in a way that invalidates existing annotations"

As far as "redefining" I think that is given as an example of the bad
> implications of changing the meaning; it invalidates the meaning of
> existing annotations.
>

Both these points were in the section
http://wiki.geneontology.org/index.php/Principles_for_term_obsoletion
My point was that I thought we *never* change the denotation of a term by
changing a definition.
If the denotation is the same and rewriting clarifies what it means then it
would fall under their example of not deprecating.

Changes in term label or definition that do not alter the meaning of the
term do not usually lead to obsoletion.
The fact that a term has incorrect annotations associated does not usually
lead to term obsoletion; ideally the database that submitted the
annotations should be informed of the error instead.

Alan

**Comment by matentzn** [comment]
Can we have the EWG draft a depreciation policy? Where would it be included on the site? @nataled @nlharris

**Comment by nataled** [comment]
We'll take it up. I imagine this would fit in a principle discussed in the early days of EWG regarding "stability of term meaning" (principle 19).

**Comment by cmungall** [comment]
any update on this? Is this even EWG, I think this is an OMO issue

**Comment by nataled** [comment]
@cmungall it is within the scope of the EWG in that there's a new principle here (which, IIRC, arose from discussions of a different principle.

**Comment by nataled** [comment]
For tracking purposes, the relevant ticket for the proposed principle 19 is #964.
